pkg/build/stages/archive_storage_reader.go

```diff
@@ -0,0 +1,96 @@
+package stages
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"os"
+)
+
+type ArchiveStorageReader interface {
+	String() string
+	ReadArchiveStage() ([]byte, error)
+}
+
+type ArchiveStorageFileReader struct {
+	Path string
+}
+
+func NewArchiveStorageFileReader(path string) *ArchiveStorageFileReader {
+	return &ArchiveStorageFileReader{
+		Path: path,
+	}
+}
+
+func (reader *ArchiveStorageFileReader) String() string {
+	return reader.Path
+}
+
+func (reader *ArchiveStorageFileReader) ReadArchiveStage() ([]byte, error) {
+	treader, closer, err := reader.openForReading()
+	defer closer()
+
+	if err != nil {
+		return nil, fmt.Errorf("unable to open bundle archive: %w", err)
+	}
+
+	b := bytes.NewBuffer(nil)
+
+	for {
+		header, err := treader.Next()
+		if err == io.EOF {
```

Comment:
Are you sure that this logic correct? tar.Reader always returns EOF in the end of reading.